src/pages/main.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { getBooks } from "../api/books";
+
+import Header from "../component/header";
+import Product from "../component/product";
+
+const Block = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  margin-left: 6%;
+  margin-right: 6%;
+  margin-top: 6%;
+`;
+
+function MainPage() {
```

Comment:
Please use this way to create functional components to take advantage of TypeScript and object destructing to get the props

```ts
const Component : React.FC<ComponentProps> = ({prop1, prop2, ...etc}) => {}
```

Not just here but everywhere